cmd/compute-domain-controller/resourceclaimtemplate.go

```diff
@@ -166,11 +175,15 @@ func (m *BaseResourceClaimTemplateManager) Create(ctx context.Context, templateP
 		return nil, fmt.Errorf("error creating ResourceClaimTemplate: %w", err)
 	}
 
+	// Add the newly created ResourceClaimTemplate to the mutation cache
+	// This ensures subsequent calls will see it immediately
+	m.mutationCache.Mutation(rct)
+
 	return rct, nil
 }
 
 func (m *BaseResourceClaimTemplateManager) Delete(ctx context.Context, cdUID string) error {
-	rcts, err := getByComputeDomainUID[*resourceapi.ResourceClaimTemplate](ctx, m.informer, cdUID)
+	rcts, err := getByComputeDomainUID[*resourceapi.ResourceClaimTemplate](ctx, m.mutationCache, cdUID)
```

Comment:
Why did you decide to not delete the item from the mutation cache before line 208 (if the `Delete()` was successful)?

Reply:
Gotcha.